### DISCUSSION TITLE
Idea inbox for extensions / plugins

### DISCUSSION THREAD

**Discussion by srid** [srid/emanote#83] [Ideas]
Dedicating a topic where people can add not-yet-fully-fleshed-out things that Emanote can support as an optional extension or a plugin. Each idea should be their own reply below.

**Comment by srid** [comment]
**Additional marks via emojis**

![image](https://user-images.githubusercontent.com/3998/123286168-af5d8480-d4db-11eb-88e2-a248fdf243e6.png)

See https://github.com/FilipVerhaeghe/emoji-review for demo in VSCode

**Comment by srid** [comment]
**Drawing graphs and figures from Markdown**

![image](https://user-images.githubusercontent.com/3998/123286423-e59b0400-d4db-11eb-90e1-bda8cf3533e2.png)

See https://github.com/srid/neuron/issues/228#issuecomment-670290253 (screenshot from @b0o)

**Reply by srid** [reply]
Or just support Mermaid: https://mermaid-js.github.io/mermaid/#/

**Reply by srid** [reply]
This now working: https://note.ema.srid.ca/tips/js/mermaid

EDIT: "working" ... as in, a basic version using client-side JS. I imagine we could also have first-class support for Mermaid, which would also allow us to support wiki-links inside Markdown notes.

**Comment by srid** [comment]
Markdown outline -> Mindmaps

- https://markmap.js.org/
- https://playground.diagram.codes/d/mind-map

**Comment by vcavallo** [comment]
well that was fast!

**Comment by srid** [comment]
**User-written plugins** (meta)

Using a runnable-in-Haskell language like Carp: https://github.com/carp-lang/Carp

User adds `plugins/foo.carp` to their notebook, and it gets loaded and used automatically, in live-server and static site generator. Plugins would have full access to the Emanote API, and be able to run code at various hook points (eg: to transform Pandoc AST programmatically).

**Comment by srid** [comment]
**Task tracker**

See https://github.com/srid/emanote/discussions/181

**Comment by j-steinbach** [comment]
Extremely far in the future, and way out of scope: Provide ways to fetch data from [Wikidata](https://www.wikidata.org/wiki/Wikidata:Main_Page). 

Example: I mention a movie and can display its year/country of origin. Or mention a painter and get a short bio. Or even their [paintings](https://commons.wikimedia.org/w/index.php?search=Van+Gogh+works+by+date&title=Special:MediaSearch&type=image).

I imagine this to work similar to _widgets_ or [embedding](https://note.ema.srid.ca/demo/embed),

**Comment by j-steinbach** [comment]
Flash cards ([Anki](https://apps.ankiweb.net/)/[Mochi](https://mochi.cards/) support).

It would be useful to be able to read/display flash cards from markdown notes. That means embedding them and hiding the backside, which can then be "flipped" over.

 Problem: There are a bunch of extensions with different syntax, but their baseline seems to be the `h2` tags. [1](https://github.com/ashlinchak/mdanki) [2](https://github.com/lukesmurray/markdown-anki-decks) [3](https://github.com/benwr/ankdown) [4](https://github.com/Pradhyo/anki-markdown-notes)


(Generation of flash cards should be done with a separate tool)

---

Note: This idea is not yet well thought out, but I want to mention it before I forget again. And maybe someone else gets inspiration.

**Comment by vcavallo** [comment]
I built a sort of crappy but working bash script that lets me symlink .md
files from anywhere on the filesystem to be used as anki/flashcards in the
program. they’re “flippable”, too.

My two cents: The spaced repetition algorithm and other features needed to
make this work is too much for emanote. BUT: having the notes live _in_
your ema library makes a lot of sense - which is why I made my script and
keep my “cards” in my zettelkasten.

here it is:
https://github.com/vcavallo/plaintext-spaced-repetition-cli
I wanted it to have near zero dependencies, which is why i didn’t make a
much better version in a more capable language.

On Mon, Oct 25, 2021 at 7:37 AM j-steinbach ***@***.***>
wrote:

> Flash cards (Anki <https://apps.ankiweb.net/>/Mochi <https://mochi.cards/>
> support).
>
> It would be useful to be able to read/display flash cards from markdown
> notes. That means embedding them and hiding the backside, which can then be
> "flipped" over.
>
> Problem: There are a bunch of extensions with different syntax, but their
> baseline seems to be the h2 tags. 1 <https://github.com/ashlinchak/mdanki>
> 2 <https://github.com/lukesmurray/markdown-anki-decks> 3
> <https://github.com/benwr/ankdown> 4
> <https://github.com/Pradhyo/anki-markdown-notes>
>
> (Generation of flash cards should be done with a separate tool)
> ------------------------------
>
> Note: This idea is not yet well thought out, but I want to mention it
> before I forget again. And maybe someone else gets inspiration.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/srid/emanote/discussions/83#discussioncomment-1531595>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYNQ36K4B3SLR3K7L3Y4FDUIU6RLANCNFSM47ICJGEQ>
> .
>
-- 
Vinney Cavallo
Founder/Lead Developer, exnil.io
